src/jailer/src/resource_limits.rs

```diff
@@ -98,6 +98,17 @@ impl ResourceLimits {
         // Set limit on number of file descriptors.
         ResourceLimits::set_limit(Resource::RlimitNoFile, self.no_file)?;
 
+        // Resize the file descriptor table to its maximal possible size, to ensure that
+        // firecracker will not need to reallocate it later. If the file descriptor table
+        // needs to be reallocated (which by default happens once more than 64 fds exist,
+        // something that happens for reasonably complex microvms due to each device using
+        // a multitude of eventfds), this can incur a significant performance impact (it
+        // was responsible for a 30ms-70ms impact on snapshot restore times).
+        // SAFETY: Duplicating stdin is safe
+        if unsafe { libc::dup2(0, self.no_file as libc::c_int - 1) } < 0 {
```

Comment:
Shouldn't new fd be closed after `dup2`? There is no point in keeping it after table is resized.

Reply:
I think @ShadowCurse is right. Unless we close the duped, we'll be keeping 2047 copies of stdin fd around for no reason.

Reply:
Ahhh, you're right. I thought it would get auto-closed by `sanitize_process`, but that called before the resource limits are setup, so it does indeed leave it open (only a single copy though, `dup2` does the same as `dup`, only that we get to choose what number our duplicate should have)

Reply:
Done :)

Reply:
That's wrong. We will not be keeping 2047 copies of the stdin, just one. But yes, we could close it.